app/models/spree/shipment.rb

```diff
@@ -116,7 +116,22 @@ def refresh_rates
 
       # The call to Stock::Estimator below will replace the current shipping_method
       original_shipping_method_id = shipping_method.try(:id)
-      self.shipping_rates = OrderManagement::Stock::Estimator.new(order).shipping_rates(to_package)
+
+      estimator = OrderManagement::Stock::Estimator.new(order)
+      distributor_shipping_rates = estimator.shipping_rates(to_package)
+
+      if original_shipping_method_id.present? &&
+         distributor_shipping_rates.map(&:shipping_method_id)
+             .exclude?(original_shipping_method_id)
+        cost = estimator.__send__(:calculate_cost, shipping_method, to_package)
+        unless cost.nil?
+          original_shipping_rate = shipping_method.shipping_rates.new(cost: cost)
+          self.shipping_rates = distributor_shipping_rates + [original_shipping_rate]
+          self.selected_shipping_rate_id = original_shipping_rate.id
+        end
+      else
+        self.shipping_rates = distributor_shipping_rates
+      end
```

Comment:
I noticed a call to a private method with `__send__`. This was a hint for me: I think the purpose of this code is to re-implement `OrderManagement::Stock::Estimator` in a specific way. 

I'm wondering if the Estimator class should be updated. Would this implementation be safe to apply directly in Estimator? I see that it is called from perhaps one other place: `Order#create_proposed_shipments` (though I didn't search extensively). 

Otherwise, perhaps this should be an alternate method next to `Estimator#shipping_rates`. Or at the very least, `calculate_cost` could be promoted to a public method.

Reply:
I didn't want to copy-and-paste the code from the estimator to avoid potential issues in case the logic is updated.
I'll promote `calculate_cost` to a public method.

Reply:
Okay, I'm still concerned that this code belongs in `Estimator`, but let's go ahead like this for now.

Reply:
I guess we should have moved back to 'In Dev' while under discussion ;)

I think this is still a good idea. Would you be confident to try it in a new PR?

Reply:
I'll do that in a different PR :)

Reply:
Sorry, I've missed these new messages - went ahead, tested and merged :grimacing: